tools/toolchains/__init__.py

```diff
@@ -577,7 +582,7 @@ def _add_dir(self, path, resources, base_path, exclude_paths=None):
                     # Ignore toolchain that do not match the current TOOLCHAIN
                     (d.startswith('TOOLCHAIN_') and d[10:] not in labels['TOOLCHAIN']) or
                     # Ignore .mbedignore files
-                    self.is_ignored(join(dir_path,"")) or
+                    self.is_ignored(join(relpath(root, base_path), d,"")) or
```

Comment:
I realize this was here before, but do you know the path is being joined with `""`?